__tests__/__unit__/DatasetTree.unit.test.ts

```diff
@@ -766,6 +766,70 @@ describe("DatasetTree Unit Tests", () => {
 
     });
 
+    it("tests the dataset filter prompt credentials, favorite route", async () => {
+        showQuickPick.mockReset();
+        showInputBox.mockReset();
+        const sessionwocred = new Session({
+            user: "",
+            password: "",
+            hostname: "fake",
+            port: 443,
+            protocol: "https",
+            type: "basic",
+        });
+        const sessNode = new ZoweNode("sestest", vscode.TreeItemCollapsibleState.Expanded, null, session);
+        sessNode.contextValue = extension.DS_SESSION_CONTEXT + extension.FAV_SUFFIX;
+        const dsNode = new ZoweNode("[testSess2]: node", vscode.TreeItemCollapsibleState.Expanded, sessNode, sessionwocred);
+        dsNode.contextValue = extension.DS_SESSION_CONTEXT + extension.FAV_SUFFIX;
+        testTree.mSessionNodes.push(dsNode);
+        const dsNode2 = new ZoweNode("testSess2", vscode.TreeItemCollapsibleState.Expanded, sessNode, sessionwocred);
+        dsNode2.contextValue = extension.DS_SESSION_CONTEXT + extension.FAV_SUFFIX;
+        testTree.mSessionNodes.push(dsNode2);
+        getConfiguration.mockReturnValue({
+            persistence: true,
+            get: (setting: string) => [
+                "[test]: brtvs99.public1.test{pds}",
+                "[test]: brtvs99.test{ds}",
+                "[test]: brtvs99.fail{fail}",
+                "[test]: brtvs99.test.search{session}",
+                "[test]: brtvs99.test.*{session}",
+            ],
+            update: jest.fn(()=>{
+                return {};
+            })
+        });
+        Object.defineProperty(Profiles, "getInstance", {
+            value: jest.fn(() => {
+                return {
+                    allProfiles: [{name: "firstName", profile: {user:undefined, password: undefined}}, {name: "secondName"}],
+                    defaultProfile: {name: "firstName"},
+                    loadNamedProfile: jest.fn(()=> {
+                        return null;
+                    }),
+                    promptCredentials: jest.fn(()=> {
+                        return ["", "", ""];
+                    }),
+                };
+            })
+        });
+
+        const spyMe = new DatasetTree();
+        Object.defineProperty(spyMe, "datasetFilterPrompt", {
+            value: jest.fn(() => {
+                return {
+                    tempNode: dsNode2,
+                    mSessionNodes: {Session: {ISession: {user: "", password: "", base64EncodedAuth: ""}}}
+                };
+            })
+        });
+
+        testTree.initialize(Logger.getAppLogger());
```

Comment:
Should we expect the error on the try-catch we just added to be shown in this case?
If it doesn't happen, maybe it's because the loadNamedProfile function is returning null instead of throwing an error?